osquery/tables/system/posix/ssl_cert.cpp

```diff
@@ -0,0 +1,199 @@
+  r["issued_organization"] = std::string(temp);
+
+  X509_NAME_get_text_by_NID(
+      subject_name, NID_organizationalUnitName, temp, SHORT_STR);
+  r["issued_organization_unit"] = std::string(temp);
+
+  ASN1_INTEGER* serial = X509_get_serialNumber(cert);
+  BIGNUM* bn = ASN1_INTEGER_to_BN(serial, NULL);
+  char* dec_str = BN_bn2dec(bn);
+  r["issued_serial_number"] = std::string(dec_str);
+  BN_free(bn);
+  OPENSSL_free(dec_str);
+
+  // set certificate issuer information
+  X509_NAME* issuer_name = X509_get_issuer_name(cert);
+  X509_NAME_get_text_by_NID(issuer_name, NID_commonName, temp, SHORT_STR);
+  r["issuer_cn"] = std::string(temp);
+
+  X509_NAME_get_text_by_NID(issuer_name, NID_organizationName, temp, SHORT_STR);
+  r["issuer_organization"] = std::string(temp);
+
+  X509_NAME_get_text_by_NID(
+      issuer_name, NID_organizationalUnitName, temp, SHORT_STR);
+  r["issuer_organization_unit"] = std::string(temp);
+
+  // set period of validity
+  ASN1_TIME* valid_from = X509_get_notBefore(cert);
+  ASN1_TIME* valid_to = X509_get_notAfter(cert);
+  BIO* b = BIO_new(BIO_s_mem());
+
+  ASN1_TIME_print(b, valid_from);
+  BIO_gets(b, temp, SHORT_STR);
+  r["valid_from"] = std::string(temp);
+
+  ASN1_TIME_print(b, valid_to);
+  BIO_gets(b, temp, SHORT_STR);
+  r["valid_to"] = std::string(temp);
+  BIO_free(b);
+
+  // set sha 256 & 1 fingerprint
+  EVP_MD* digest = (EVP_MD*)EVP_sha256();
+  unsigned len = SHORT_STR;
+  X509_digest(cert, digest, (unsigned char*)temp, &len);
+  char sha_text[SHORT_STR + 1];
+  memset(sha_text, 0, sizeof(sha_text));
+
+  for (unsigned i = 0; i < len; i++) {
+    char byt[5];
+    sprintf(byt, "%02X%c", temp[i], (i + 1 == len) ? '\0' : ':');
+    strcat(sha_text, byt);
+  }
+
+  r["sha256_fingerprint"] = std::string(sha_text);
+  digest = (EVP_MD*)EVP_sha1();
+  X509_digest(cert, digest, (unsigned char*)temp, &len);
+  memset(sha_text, 0, sizeof(sha_text));
+
+  for (unsigned i = 0; i < len; i++) {
+    char byt[5];
+    sprintf(byt, "%02X%c", temp[i], (i + 1 == len) ? '\0' : ':');
+    strcat(sha_text, byt);
+  }
+
+  r["sha1_fingerprint"] = std::string(sha_text);
+
+  return;
+}
+
+static void getSslCert(const char* issued_cname, QueryData& results) {
+  char* def_cert_dir;
+
+  // common certificate directory for POSIX
+  const std::string cert_dir_list[6] = {"/etc/pki/tls/certs",
+                                        "/etc/ssl/certs",
+                                        "/etc/ssl",
+                                        "/etc/pki/ca-trust/extracted/pem",
+                                        "/usr/local/share/certs",
+                                        "/etc/openssl/certs"};
+
+  def_cert_dir = getenv(X509_get_default_cert_dir_env());
+
+  if (!def_cert_dir) {
+    def_cert_dir = (char*)X509_get_default_cert_dir();
+  }
+
+  fs::path full_path(fs::initial_path<fs::path>());
+
+  full_path = fs::system_complete(fs::path(def_cert_dir));
```

Comment:
Any reason for the `fs::system_complete` here? Why not just `auto full_path = fs::path(def_cert_dir);`?